testing/bsg_rect_to_polar/bsg_hypotenuse_script.py

```diff
@@ -0,0 +1,222 @@
+import math, sys
+
+def lookup_compute(posprec, precision_result):
+    lookup=[]
+    for j in range(0,posprec+1):
+        m=(math.atan2(1,2**j)*180)/math.pi
+        lookup.append(format(round(m*(2**precision_result)),'x'))
+    return lookup
+
+def constant_compute(posprec, ansbitlen):
+    const=1
+    for i in range(0,posprec):
+        comp = math.cos(math.atan2(1,2**i))
+        const=const*comp
+    const = format(round(const*(2**(ansbitlen-1))),'x')
+    return const
+
+def signed_constant(ansbitlen):
+    const = "1"
+    for i in range(0,ansbitlen-1):
+        const = const + str(0)
+    return (hex(int(const, 2)))
+
+def signed_constant2(ansbitlen):
+    const = ""
+    for i in range(0,ansbitlen):
+        const = const + str(1)
+    return (hex(int(const, 2)))
+
+def bsg_sine_cosine_init(angbitlen, ansbitlen, posprec):
+    print('''
+    module bsg_cordic_hypotenuse #(precis_p = %(p)d, ang_width_p = %(g)d, ans_width_p = %(s)d, scale_width_p = %(sw)d)
+    (
+    input clk_i
+    ,input signed [ans_width_p-1:0] x_i
+    ,input signed [ans_width_p-1:0] y_i
+    ,input ready_i
+    ,input val_i
+    ,output signed [ans_width_p-1:0] mag_o
+    ,output signed [ang_width_p-1:0] angl_o
+    ,output ready_o
+    ,output val_o
+    );
+    
+    logic [ans_width_p-1:0] x_set, y_set;
+    logic signed [ans_width_p-1:0] x_in, y_in;
+    logic val_in;
+    logic signed [precis_p+1:0][ans_width_p-1:0] x ;
+    logic signed [precis_p+1:0][ans_width_p-1:0] y ;
+    logic signed [precis_p+1:0][ang_width_p-1:0] ang ;
+    logic [precis_p+1:0] val ;
+    logic [precis_p+1:0] switch ;
+    logic [precis_p+1:0] quad_x ;
+    logic [precis_p+1:0] quad_y ;
+    logic signed [precis_p:0][ans_width_p-1:0] x_ans ;
+    logic signed [precis_p:0][ans_width_p-1:0] y_ans ;
+    logic signed [precis_p:0][ang_width_p-1:0] ang_ans ; 
+    logic [precis_p:0] val_ans ;
+    logic [precis_p:0] switch_ans ;
+    logic [precis_p:0] quad_x_ans ;
+    logic [precis_p:0] quad_y_ans ;
+    logic val_out;
+    wire stall_pipe = val_o & (~ready_i);
+    
+    ''' %{'s':ansbitlen, 'g':angbitlen, 'p':posprec, 'sw':precisionbitlen})
+    return
+
+def lookup_initialization(posprec, angbitlen, result):
```

Comment:
Do you mean the lookup_initialization function? It generates the lookup table for each cordic stage which will be different for each function and will depend upon the number of stages input by the user. In case of some functions this for loop will iterate upon single parameter "posprec" and for others, it will be "posprec + negprec".